jerry-core/ecma/base/ecma-globals.h

```diff
@@ -196,6 +196,25 @@ typedef int32_t ecma_integer_value_t;
 typedef uintptr_t ecma_external_pointer_t;
 
 /**
+ * Representation for native handle package.
+ */
+
+typedef struct
+{
+  ecma_external_pointer_t handle_p; /**< points to the external native object */
+  ecma_external_pointer_t info_p; /**< type info of the native handle */
+} ecma_native_handle_package_t;
+
+/**
+ * Representation of the native handle package for deprecated api.
+ */
+typedef struct
+{
+  ecma_native_handle_package_t base; /**< the package's info_p is NULL */
+  ecma_external_pointer_t free_cb; /**< free callback of the native handle */
+} ecma_native_handle_package_old_t;
```

Comment:
The old API is now called `jerry_[g|s]et_object_native_handle_deprecated()` instead. Therefore, it's technically not compatible because any code that relies on the old API would have to rename the calls (add `..._deprecated`).